klippy/extras/sht4x.py

```diff
@@ -0,0 +1,158 @@
+# SHT4x i2c temperature sensors
+#
+# Copyright (C) 2025 Milzo <overchecking@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+import logging
+from . import bus
+
+# SHT4x I2C Commands
+DEFAULT_ADDR = 0x44
+RESET = 0x94
+
+# Measurement command (high precision)
+MEASUREMENT_CMD = 0xFD
+
+# Measurement timing constant
+MEASUREMENT_TIME = 0.009  # 9ms
+RESET_TIME = 0.001  # 1ms
+
+class SHT4X:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.name = config.get_name().split()[-1]
+        self.reactor = self.printer.get_reactor()
+
+        # I2C setup
+        addr = config.getint('i2c_address', DEFAULT_ADDR)
+        self.i2c = bus.MCU_I2C_from_config(config, default_addr=addr)
+        self.mcu = self.i2c.get_mcu()
+
+        # Sensor state
+        self.temp = 0
+        self.humidity = 0
+        self._error_count = 0
+        self.sensor_ready = False
+
+        # Register with Klipper
+        self.printer.add_object("sht4x " + self.name, self)
+        self.printer.register_event_handler(
+            "klippy:connect", self.handle_connect)
+
+    def handle_connect(self):
+        """Initialize sensor"""
+        try:
+            self.reset()
+            if self._test_measurement():
+                self.sensor_ready = True
+                logging.info("SHT4X: Sensor ready")
+                self.sample_timer = self.reactor.register_timer(
+                    self.sample_sensor)
+                self.reactor.update_timer(self.sample_timer, self.reactor.NOW)
+            else:
+                logging.error("SHT4X: Sensor test failed")
+        except Exception as e:
+            logging.error("SHT4X: Init failed: %s" % str(e))
+
+    def _test_measurement(self):
+        """Quick measurement test"""
+        try:
+            recv = self.get_measurements()
+            return self._validate_crc(recv)
+        except Exception:
+            return False
+
+    def _validate_crc(self, data):
+        """Validate CRC for both temp and humidity"""
+        temp_crc = data[2]
+        humidity_crc = data[5]
+        return (temp_crc == self._crc8(data[0:2]) and
+                humidity_crc == self._crc8(data[3:5]))
+
+    def setup_minmax(self, min_temp, max_temp):
```

Comment:
Why do we disable directly asked error conditions?
I do set up the chamber, and I do run the heater.
I do want the machine to die if the sensor report me data outside of the safe range.